__tests__/integration/atomic-swap.test.js

```diff
@@ -0,0 +1,305 @@
+    loggers.test.insertLineToLog('atomic-swap[1TK P2PKH]: push', { body: response.body });
+    expect(response.body.success).toBe(true);
+
+    await TestUtils.pauseForWsUpdate();
+  });
+
+  it('should exchange multiple tokens between 2 wallets', async () => {
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        send_tokens: [{ token: tokenTx1.hash, value: 5 }, { token: '00', value: 10 }],
+        receive_tokens: [{ token: tokenTx2.hash, value: 15 }],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[2TK P2PKH]: proposal', { body: response.body });
+
+    let { data } = response.body;
+
+    expect(response.body.isComplete).toBe(false);
+
+    // wallet2 updates the proposal
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        partial_tx: data,
+        receive_tokens: [{ token: tokenTx1.hash, value: 5 }, { value: 10 }],
+        send_tokens: [{ token: tokenTx2.hash, value: 15 }],
+      })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    loggers.test.insertLineToLog('atomic-swap[2TK P2PKH]: update proposal', { body: response.body });
+    expect(response.body.isComplete).toBe(true);
+
+    data = response.body.data;
+
+    // wallet1: sign data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/get-my-signatures')
+      .send({ partial_tx: data })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[2TK P2PKH]: get signatures', { body: response.body });
+    expect(response.body).toEqual({
+      success: true,
+      isComplete: false,
+      signatures: expect.any(String),
+    });
+    const signature = response.body.signatures;
+
+    // wallet2: sign-and-push data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/sign-and-push')
+      .send({ partial_tx: data, signatures: [signature] })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    loggers.test.insertLineToLog('atomic-swap[2TK P2PKH]: push', { body: response.body });
+    expect(response.body.success).toBe(true);
+
+    await TestUtils.pauseForWsUpdate();
+  });
+
+  it('should allow multisig wallets to participate', async () => {
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        send_tokens: [{ value: 200 }], // HTR
+        receive_tokens: [
+          { token: tokenTx2.hash, value: 50 },
+          { token: tokenTx1.hash, value: 50 },
+        ],
+      })
+      .set({ 'x-wallet-id': walletMultisig.walletId });
+    loggers.test.insertLineToLog('atomic-swap[P2SH]: proposal', { body: response.body });
+    expect(response.body.isComplete).toBe(false);
+
+    let { data } = response.body;
+
+    // wallet1 updates the proposal
```

Comment:
What happens here? It's important to describe in the comment
Will the proposal be completed substituted? Or it will merge somehow?

Reply:
Changed the comments to reflect which actions are being added to the proposal.